imqsauth/http.go

```diff
@@ -1109,11 +1111,19 @@ func createGroupsChangedNotificationMessage(messageContent string) string {
 }
 
 func httpHandlerSetPassword(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
-	userId, getUserIdErr := getUserId(r)
-	if getUserIdErr != nil {
-		authaus.HttpSendTxt(w, http.StatusBadRequest, getUserIdErr.Error())
+	identity, userId, err := getUserIdOrIdentity(r)
+	if err != nil {
+		authaus.HttpSendTxt(w, http.StatusBadRequest, err.Error())
 		return
 	}
+	if identity != "" {
```

Comment:
Can you add a comment here explaining what's going on? The code flow just seems strange.